torch/_inductor/codegen/cpp_flex_attention_template.py

```diff
@@ -1017,13 +1457,26 @@ def render(  # type: ignore[override,return]
             mask_buf_name=self.mask_buf_name,
             score_buf_idx=self.score_buf_idx,
             mask_buf_idx=self.mask_buf_idx,
+            partition_size=self.partition_size,
         )
         with contextlib.ExitStack() as stack:
             for buf in self.fake_buffers:
                 stack.enter_context(
                     patch.object(V.graph, "get_dtype", self._fake_get_dtype(buf))
                 )
-            return self._template_from_string(FLEX_ATTENTION_TEMPLATE).render(**options)
+            if (
+                query.data.data.layout.size[2] == 1
```

Comment:
we may double check the condition for selecting the flash decoding path, ref: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/kernel/flex/flex_decoding.py#L34

Reply:
Thanks! A function to choose flex template is added.